webapp/src/pages/Settings.tsx

```diff
@@ -207,6 +217,17 @@ const Settings: React.FC = () => {
     Partial<AzimuthConfig>
   >({});
 
+  React.useEffect(() => {
+    if (defaultConfig && language !== undefined) {
```

Comment:
To check if we got a new `defaultConfig`, we can check if its `language` is different than the current one. For that, we will need to move initializing `resultingConfig` right before that:
```suggestion
  const resultingConfig = Object.assign({}, config, partialConfig);

  React.useEffect(() => {
    if (defaultConfig && defaultConfig.language !== resultingConfig.language) {
```

Reply:
Was there a problem moving `resultingConfig` before the `React.useEffect()`? Instead of doing all that `partialConfig.<...> ?? config.<...>` logic?

Reply:
Right!. I can explain the issue of using"resultingConfig."

If I add "resultingConfig" to the use effect dependency array, I get the warning "Warning: Maximum update depth exceeded. This can happen when a component calls setState inside use effect, but use effect either doesn't have a dependency array or one of the dependencies changes on every render." in the console. 

If I don't add it to the dependency array, the runtime error is gone, but I get compile time warning.

Reply:
If "one of the dependencies changes on every render", I guess that's because your condition for calling `setPartialConfig` had an issue.